### DISCUSSION TITLE
Is GetFormalCharge for mol or atom?

### DISCUSSION THREAD

**Discussion by arunraja-hub** [rdkit/rdkit#7196] [Q&A] [answered]
Is GetFormalCharge used to get the total charge of a molecule or the atomic charge of an atom?

According to https://www.rdkit.org/docs/source/rdkit.Chem.rdchem.html#rdkit.Chem.rdchem.Atom.GetFormalCharge, GetFormalCharge is for the atom object but I ran into the following error when I try to use GetFormalCharge for an atom object. So is it for a mol object?

```
Boost.Python.ArgumentError: Python argument types in
    rdkit.Chem.rdmolops.GetFormalCharge(Atom)
did not match C++ signature:
    GetFormalCharge(RDKit::ROMol)

```

**Comment by dehaenw** [comment, marked as answer]
Chem.GetFormalCharge(mol) gets the formal charge of the entire molecule mol
a.GetFormalCharge() gets the formal charge of the atom a
see below codeblock for an example
```python
from rdkit import Chem
mol=Chem.MolFromSmiles("[NH3+][C@H](C(=O)[O-])CCCC[NH3+]")

print("formal charge of the entire molecule is:",Chem.GetFormalCharge(mol))
print("formal charge of every individual atom is:")
for a in mol.GetAtoms():
    print(a.GetFormalCharge())
```

**Reply by dehaenw** [reply]
apologies, i didn\t see you are using C++ and not python. But the same principle applies.

**Reply by arunraja-hub** [reply]
Thanks for the clarification, no I am using Python so its fine

**Reply by arunraja-hub** [reply]
A related question: what is the difference between GetAtomicCharges (https://www.rdkit.org/docs/source/rdkit.Chem.rdEHTTools.html#rdkit.Chem.rdEHTTools.EHTResults.GetAtomicCharges) and Get Formal Charge when used for an atom?

**Reply by dehaenw** [reply]
Formal charge is integer and dependent on how you assign them to each atom (via your input, or your protomer enumeration model or whatever). EHT will calculate partial charges for every atom based  on the extended huckel method. In case you want fast partial charges, the typical way to do it is calculating Gasteiger Charges with https://www.rdkit.org/docs/source/rdkit.Chem.rdPartialCharges.html

here's a code block showing a minimal example of charges calculated by EHT method
```python
from rdkit import Chem
from rdkit.Chem import rdEHTTools
from rdkit.Chem import AllChem


m = Chem.MolFromSmiles("c1cocc1CC[NH3+]")
m = Chem.AddHs(m)
AllChem.EmbedMolecule(m)
_, res = rdEHTTools.RunMol(m)
static_chgs = res.GetAtomicCharges()
print(static_chgs)
```

**Reply by arunraja-hub** [reply]
@dehaenw I see. Is there a partial charge method in RDKit where all the partial charges generated sum up to the total charge (integer) of the molecule? When I summed up Gasteiger partial charges (using the default 12 iterations) to get the total molecular charge, the sums for some molecules were not whole numbers but values like 0.7, 1.2 etc. ...
